src/libraries/System.Collections.Immutable/src/System/Linq/ImmutableArrayExtensions.cs

```diff
@@ -187,28 +187,39 @@ public static bool SequenceEqual<TDerived, TBase>(this ImmutableArray<TBase> imm
         /// <typeparam name="TBase">The type of element contained by the collection.</typeparam>
         public static bool SequenceEqual<TDerived, TBase>(this ImmutableArray<TBase> immutableArray, IEnumerable<TDerived> items, IEqualityComparer<TBase>? comparer = null) where TDerived : TBase
         {
-            Requires.NotNull(items, nameof(items));
+            if (items is ICollection<TBase> itemsCol)
+            {
+                immutableArray.ThrowNullRefIfNotInitialized();
+                return Enumerable.SequenceEqual(immutableArray.array!, itemsCol, comparer);
```

Comment:
```suggestion
                return Enumerable.SequenceEqual(immutableArray.array, itemsCol, comparer);
```

Not necessary after https://github.com/dotnet/runtime/pull/119898